src/components/CallingCodePicker/CallingCodePicker.tsx

```diff
@@ -65,13 +59,13 @@ const CallingCodePicker: React.FC<ICallingCodePickerProps> = ({
 }) => {
   const [searchValue, setSearchValue] = useState<string>('');
   const [isPickerOpen, setIsPickerOpen] = useState<boolean>(false);
-  const [selectedCountry, setSelectedCountry] = useState<ICountry>(initialCountry);
+  const [selectedCountry, setSelectedCountry] = useState<ICountry | undefined>(undefined);
   const [countriesData, setCountriesData] = useState<ICountry[]>(countries);
-
-  const alphaCodeWithFallback = selectedCode || initialCountry.alpha;
+  const localCountryCode = RNLocalize.getCountry();
```

Comment:
Getting the user's country (based on its device locale) with [react-native-localize](https://github.com/zoontek/react-native-localize) package